File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
from collections import deque

def solution(s):
    dq = deque(s)
    count = 0
    while dq:
        value = dq.popleft()
        if value == '(':
            count += 1
        else:
            count -= 1

        if count < 0:
            return False
    if count == 0:
        return True
    else:
        return False

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.swift
import Foundation

 
func solution(_ s:String) -> Bool {
    var openBraceCount = 0
    
    for c in s {
        if openBraceCount > 0 && c == ")" {
            openBraceCount -= 1
        } else {
            openBraceCount += 1
        }
    }
 
    return openBraceCount == 0
}